sway-core/src/language/module.rs

```diff
@@ -5,3 +5,72 @@ use sway_types::Ident;
 /// If an alias was given to the `mod`, this will be the alias. If not, this is the submodule's
 /// library name.
 pub type ModName = Ident;
+
+pub trait HasModule<T>
+where
+    T: HasSubmodules<Self>,
+    Self: Sized,
+{
+    /// Returns the module of this submodule.
+    fn module(&self) -> &T;
+}
+
+pub trait HasSubmodules<E>
+where
+    E: HasModule<Self>,
+    Self: Sized,
+{
+    /// Returns the submodules of this module.
+    fn submodules(&self) -> &Vec<(ModName, E)>;
```

Comment:
Any reason to return a `&Vec<(ModName, E)>` instead of `&[(ModName, E)]` ?

Reply:
No reason, updated